### PR TITLE
Run Kubernetes master branch unit tests as non-root unprivileged user

### PR DESCRIPTION
**Pull request by BenTheElder**
Fixes https://github.com/kubernetes/kubernetes/issues/99881 _finally_!

Thanks @PushkarJ for fixing the last problematic unit test 🎉

### CODE REVIEW

File: config/jobs/kubernetes/sig-testing/make-test.yaml [hunk 1]
```diff
@@ -14,6 +14,10 @@ presubmits:
     always_run: true
     path_alias: k8s.io/kubernetes
     spec:
+      # unit tests have no buisiness requiring root or doing privileged operations
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-testing/make-test.yaml:17]: all of these comments misspelled business

Comment (BenTheElder) [reply config/jobs/kubernetes/sig-testing/make-test.yaml]: fixed.

File: config/jobs/kubernetes/sig-testing/make-test.yaml [hunk 2]
```diff
@@ -14,6 +14,10 @@ presubmits:
     always_run: true
     path_alias: k8s.io/kubernetes
     spec:
+      # unit tests have no business requiring root or doing privileged operations
+      securityContext:
+        runAsUser: 2000
```
Comment (alvaroaleman) [inline config/jobs/kubernetes/sig-testing/make-test.yaml:19]: Does that user exist? Tests tend to do stuff in $HOME which if inexistent results in writing to / which wont work

Comment (BenTheElder) [reply config/jobs/kubernetes/sig-testing/make-test.yaml:19]: It doesn't, and they don't, unit tests should write to a tempdir and not the real $HOME.

We already had a test version of the job (`pull-kubernetes-unit-experimental`) that passes with this config.

File: config/jobs/kubernetes/sig-testing/make-test.yaml [hunk 3]
```diff
@@ -14,6 +14,10 @@ presubmits:
...
+        allowPrivilegeEscalation: false
```
Comment (vinayakankugoyal) [inline config/jobs/kubernetes/sig-testing/make-test.yaml:20]: Should we drop all capabilities and also add `runAsGroup: 2000`?

Comment (BenTheElder) [reply config/jobs/kubernetes/sig-testing/make-test.yaml:20]: probably the right call, we should start in the -experimental job first though and make sure there isn't something depending on this (and if so fix it before rolling it out to everyone's PRs)

Comment (BenTheElder) [reply config/jobs/kubernetes/sig-testing/make-test.yaml:20]: closing the loop https://github.com/kubernetes/test-infra/pull/22780

**Pull request merged**